spec/features/dashboard/protocols/index/filters_spec.rb

```diff
@@ -213,108 +224,337 @@ def visit_protocols_index_page
+        create(:project_role, identity: user, role: "very-important", project_rights: "to-party", protocol: @protocol1)
 
-      expect(@page.search_results).to have_protocols(text: "titlex")
-      expect(@page.search_results).to have_protocols(text: "xTitle")
-      expect(@page.search_results).to have_no_protocols(text: "aaa")
-    end
+        @protocol2 = create_protocol(archived: false, title: "xTitle", short_title: "Protocol2")
+        create(:project_role, identity: user, role: "very-important", project_rights: "to-party", protocol: @protocol2)
 
-    it "should match against title case insensitively" do
-      titlexProtocol = create_protocol(archived: false, title: "titlex", short_title: "Protocol1")
-      titlexProtocol.project_roles.create(identity_id: user.id, role: "very-important", project_rights: "to-party")
-      xTitleProtocol = create_protocol(archived: false, title: "xTitle", short_title: "Protocol2")
-      xTitleProtocol.project_roles.create(identity_id: user.id, role: "very-important", project_rights: "to-party")
-      aaaProtocol = create_protocol(archived: false, title: "aaa", short_title: "Protocol3")
-      aaaProtocol.project_roles.create(identity_id: user.id, role: "very-important", project_rights: "to-party")
+        @protocol3 = create_protocol(archived: false, title: "aaa", short_title: "Protocol3")
+        create(:project_role, identity: user, role: "very-important", project_rights: "to-party", protocol: @protocol3)
 
-      visit_protocols_index_page
-      @page.filter_protocols.search_field.set("title")
-      @page.filter_protocols.apply_filter_button.click
+        service_request1 = create(:service_request_without_validations, protocol: @protocol1)
+                           create(:sub_service_request, service_request: service_request1, organization: organization1, status: 'draft')
+
+        service_request2 = create(:service_request_without_validations, protocol: @protocol2)
+                           create(:sub_service_request, service_request: service_request2, organization: organization2, status: 'draft')
+
+        service_request3 = create(:service_request_without_validations, protocol: @protocol3)
+                           create(:sub_service_request, service_request: service_request3, organization: organization3, status: 'draft')
+      end
 
-      expect(@page.search_results).to have_protocols(text: "Protocol1")
-      expect(@page.search_results).to have_protocols(text: "Protocol2")
-      expect(@page.search_results).to have_no_protocols(text: "Protocol3")
+      it "should match against title case insensitively (lowercase)" do
+        visit_protocols_index_page
+        expect(@page.search_results).to have_protocols(count: 3)   
+        @page.filter_protocols.select_search(@page, "Protocol", "title")
+        @page.filter_protocols.apply_filter_button.click()
+        wait_for_javascript_to_finish
```

Comment:
How necessary is the `wait_for_javascript`? I see that the example below doesn't have it.